applications/convection_diffusion_application/python_scripts/convection_diffusion_base_solver.py

```diff
@@ -109,59 +163,7 @@ def __init__(self, model, custom_settings):
-                KratosMultiphysics.Logger.PrintWarning("::[ConvectionDiffusionBaseSolver]:: ", " W-A-R-N-I-N-G: MESH VELOCITY VARIABLE NOT DEFINED, TAKING DEFAULT", default_settings["convection_diffusion_variables"]["mesh_velocity_variable"].GetString())
-            if not custom_settings["convection_diffusion_variables"].Has("transfer_coefficient_variable"):
-                KratosMultiphysics.Logger.PrintWarning("::[ConvectionDiffusionBaseSolver]:: ", " W-A-R-N-I-N-G: TRANSFER COEFFICIENT VARIABLE NOT DEFINED, TAKING DEFAULT", default_settings["convection_diffusion_variables"]["transfer_coefficient_variable"].GetString())
-            if not custom_settings["convection_diffusion_variables"].Has("velocity_variable"):
-                KratosMultiphysics.Logger.PrintWarning("::[ConvectionDiffusionBaseSolver]:: ", " W-A-R-N-I-N-G: VELOCITY VARIABLE NOT DEFINED, TAKING DEFAULT", default_settings["convection_diffusion_variables"]["velocity_variable"].GetString())
-            if not custom_settings["convection_diffusion_variables"].Has("specific_heat_variable"):
-                KratosMultiphysics.Logger.PrintWarning("::[ConvectionDiffusionBaseSolver]:: ", " W-A-R-N-I-N-G: SPECIFIC HEAT VARIABLE NOT DEFINED, TAKING DEFAULT", default_settings["convection_diffusion_variables"]["specific_heat_variable"].GetString())
-            if not custom_settings["convection_diffusion_variables"].Has("reaction_variable"):
-                KratosMultiphysics.Logger.PrintWarning("::[ConvectionDiffusionBaseSolver]:: ", " W-A-R-N-I-N-G: REACTION VARIABLE NOT DEFINED, TAKING DEFAULT", default_settings["convection_diffusion_variables"]["reaction_variable"].GetString())
-
-        # Overwrite the default settings with user-provided parameters.
-        self.settings = custom_settings
-        self.settings.ValidateAndAssignDefaults(default_settings)
-        model_part_name = self.settings["model_part_name"].GetString()
-
-        # Set default buffer size
-        self.min_buffer_size = 1
-
-        if model_part_name == "":
-            raise Exception('Please specify a model_part name!')
-
-        # This will be changed once the Model is fully supported!
-        if self.model.HasModelPart(model_part_name):
-            self.main_model_part = self.model[model_part_name]
-            self.solver_imports_model_part = False
-        else:
-            self.main_model_part = self.model.CreateModelPart(model_part_name) # Model.CreateodelPart()
-            domain_size = self.settings["domain_size"].GetInt()
-            if domain_size < 0:
-                raise Exception('Please specify a "domain_size" >= 0!')
-            self.main_model_part.ProcessInfo.SetValue(KratosMultiphysics.DOMAIN_SIZE, domain_size)
-            self.solver_imports_model_part = True
-
-        KratosMultiphysics.Logger.PrintInfo("::[ConvectionDiffusionBaseSolver]:: ", "Construction finished")
+        return default_settings
```

Comment:
I think you also need to add the missing params here, same as I did in StructuralMechanics,  otherwise I think the derivation of solvers doesn't work properly

Reply:
There should be all looking at the differences in this merge request

Reply:
I mean this: https://github.com/KratosMultiphysics/Kratos/blob/master/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py#L123

Reply:
Thanks!